### PR TITLE
Feat/pidgin

### PR DESCRIPTION
**Pull request by zflamig**

### CODE REVIEW

File: apis_configs/pidgin_settings.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+APP_NAME='pidgin'
+def load_json(file_name):
+  return config_helper.load_json(file_name, APP_NAME)
+
+#conf_data = load_json('creds.json')
+config = app.config
+
+config['API_URL'] = 'http://peregrine-service/v0/submission/graphql/'
+#app_init(app)
+application = app
```
Comment (frickjack) [inline apis_configs/pidgin_settings.py]: Can we get rid of this, and just have the default `settings.py` in pidgin load a json file for configuration?  You can pull the `config_helper` into `pyutils`, so it's generally available ...

BTW - if this is green field - I hope we're running python 3 ...

File: gen3/bin/kube-setup-pidgin.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+#
+# Deploy pidgin into existing commons
+# This fragment is pasted into kube-services.sh by kube.tf.
+#
+
+_KUBE_SETUP_PIDGIN=$(dirname "${BASH_SOURCE:-$0}")  # $0 supports zsh
+source "${_KUBE_SETUP_PIDGIN}/../lib/kube-setup-init.sh"
```
Comment (frickjack) [inline gen3/bin/kube-setup-pidgin.sh]: Please use the same sourcing mechanism we're using in the other scripts - it avoids double-sourcing files:
```

source "${GEN3_HOME}/gen3/lib/utils.sh"
gen3_load "gen3/lib/kube-setup-init"
```

File: gen3/bin/kube-setup-secrets.sh [hunk 3]
```diff
@@ -226,6 +226,27 @@ if [[ -f "${WORKSPACE}/${vpc_name}/creds.json" ]]; then # update peregrine secre
   fi
 fi
 
+if [[ -f "${WORKSPACE}/${vpc_name}_output/creds.json" ]]; then # update pidgin secrets
+  if [ ! -d "${WORKSPACE}/${vpc_name}" ]; then
```
Comment (frickjack) [inline gen3/bin/kube-setup-secrets.sh]: This is out of date - should be `"${WORKSPACE}/${vpc_name}/creds.json` - look at the other blocks

File: apis_configs/pidgin_settings.py (hunk 1 above)
Comment (frickjack) [reply apis_configs/pidgin_settings.py]: So it looks like the only thing this file does is set 'API_URL' - you can get rid of the `config_helper` stuff, and yeah - would be cool to not even need this file ...

File: kube/services/netpolicy/networkpolicy_pidgin_templ.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+         - port: 80
+         - port: 443
+  egress:
+    - to:
+      - podSelector:
+          matchLabels:
+            app: pidgin
+      - podSelector:
+          matchLabels:
+            app: indexd
```
Comment (frickjack) [inline kube/services/netpolicy/networkpolicy_pidgin_templ.yaml]: Does 'pidgin' need to talk with indexd?  If so, then we does the settings.py only configure peregrine?

File: kube/services/pidgin/pidgin-deploy.yaml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+            name: config-helper
+        - name: cert-volume
+          secret:
+            secretName: "cert-pidgin-service"
+        - name: ca-volume
+          secret:
+            secretName: "service-ca"
+      containers:
+        - name: pidgin
+          GEN3_PIDGIN_IMAGE
```
Comment (frickjack) [inline kube/services/pidgin/pidgin-deploy.yaml]: Please add entries to all the manifests in `uc-cdis/cdis_manifest` before merging this PR

File: kube/services/pidgin/pidgin-deploy.yaml [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+          - containerPort: 443
+          volumeMounts:
+            - name: "config-volume"
+              readOnly: true
+              mountPath: "/var/www/pidgin/wsgi.py"
+              subPath: "wsgi.py"
+            - name: "config-helper"
+              readOnly: true
+              mountPath: "/var/www/pidgin/config_helper.py"
+              subPath: config_helper.py
```
Comment (frickjack) [inline kube/services/pidgin/pidgin-deploy.yaml]: Hopefully we can get rid of both wsgi.py and config_helper.py ....

File: kube/services/pidgin/pidgin-deploy.yaml [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+              subPath: "service.key"
+            - name: "ca-volume"
+              readOnly: true
+              mountPath: "/mnt/ssl/cdis-ca.crt"
+              subPath: "ca.pem"
+          imagePullPolicy: Always
+          resources:
+            limits:
+              cpu: 0.8
+              memory: 128Mi
```
Comment (frickjack) [inline kube/services/pidgin/pidgin-deploy.yaml]: Does this service use `authutils` to download fence public keys?  If so, then you need to register the certificate authority with the runtime for the code to work with self signed certificates and internal cert authorities ...

File: kube/services/pidgin/pidgin-service.yaml [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    app: pidgin
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 80 
+      name: http
+    - protocol: TCP 
+      port: 443 
+      targetPort: 443
+      name: https
```
Comment (frickjack) [inline kube/services/pidgin/pidgin-service.yaml]: Is pidgin actually listening on port 443?  With the provided SSL cert and key?

File: gen3/bin/kube-setup-pidgin.sh [hunk 9]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+#
+# Deploy pidgin into existing commons
+# This fragment is pasted into kube-services.sh by kube.tf.
+#
+
+source "${GEN3_HOME}/gen3/lib/utils.sh"
```
Comment (frickjack) [inline gen3/bin/kube-setup-pidgin.sh]: Add a call to `kube-setup-pidgin` to `kube-setup-all`

Comment (paulineribeyre) [reply gen3/bin/kube-setup-pidgin.sh]: done

**Pull request merged**